packages/synapse-react-client/src/components/CitationPopover/CitationPopover.test.tsx

```diff
@@ -0,0 +1,178 @@
+import { render, screen, waitFor, within } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import CitationPopover from './CitationPopover'
+import { useCitation } from './useCitation'
+import { createLinkAndDownload } from './CitationPopoverUtils'
+import {
+  getUseQueryErrorMock,
+  getUseQueryLoadingMock,
+  getUseQuerySuccessMock,
+} from '../../testutils/ReactQueryMockUtils'
+import { UseQueryResult } from '@tanstack/react-query'
+
+jest.mock('./useCitation', () => ({
+  useCitation: jest.fn(),
+}))
+
+jest.mock('./CitationPopoverUtils', () => ({
+  createLinkAndDownload: jest.fn(),
+}))
+
+const mockUseCitation = jest.mocked(useCitation)
+const mockCreateLinkAndDownload = jest.mocked(createLinkAndDownload)
+
+const openPopover = async () => {
+  const button = screen.getByRole('button', { name: /Cite As/i })
+  await userEvent.click(button)
+}
+
+const data =
+  '@misc{test2025,\n  title = {Some BibTeX Entry},\n  year = {2025}\n}'
+
+const mockProps = {
+  doi: 'https://doi.org/10.1234/abcd1234',
+  title: 'Some Article',
+  boilerplateText: 'Some boilerplate text',
+}
+
+describe('CitationPopover tests', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+    mockUseCitation.mockReturnValue(getUseQuerySuccessMock(data))
+  })
+
+  it('renders button', async () => {
+    render(<CitationPopover {...mockProps} />)
+    await screen.findByRole('button', { name: /Cite As/i })
+  })
+
+  it('opens popover when button is clicked and fetches citation', async () => {
+    render(<CitationPopover {...mockProps} />)
+    openPopover()
+
+    await screen.findByRole('presentation', {
+      name: /Citation options popover/i,
+    })
```

Comment:
The [`presentation` role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/presentation_role) seems to hide this from the accessibility tree, which is not what we want. Can you set the `aria-role` of this element to be `'dialog'` and query for the `'dialog'` role in these tests?